public/app/features/datasources/components/picker/DataSourceDropdown.tsx

```diff
@@ -136,7 +153,8 @@ export function DataSourceDropdown(props: DataSourceDropdownProps) {
   const onClose = useCallback(() => {
     setFilterTerm('');
     setOpen(false);
-  }, [setOpen]);
+    markerElement?.focus();
```

Comment:
that's a better place than in Escape! Because now we are never concerned with the problem of the content living inside a Portal!!